website/docs/language/resources/ephemeral/index.mdx

```diff
@@ -44,21 +44,43 @@ Use write-only arguments to securely pass temporary values to resources during a
 <CodeBlockConfig highlight="11">
 
 ```hcl
-ephemeral "random" "password" {
+ephemeral "random_password" "db_password" {
   length = 16
 }
 
-resource "aws_db_instance" "test" {
-  instance_class      = "db.t5.micro"
+resource "aws_secretsmanager_secret" "db_password" {
+  name = "db_password"
+}
+
+resource "aws_secretsmanager_secret_version" "db_password" {
+  secret_id                = aws_secretsmanager_secret.db_password.id
+  secret_string_wo         = ephemeral.random_password.db_password.result
+  secret_string_wo_version = 1
+}
+
+ephemeral "aws_secretsmanager_secret_version" "db_password" {
+  secret_id = aws_secretsmanager_secret.db_password.id
```

Comment:
```suggestion
  secret_id = aws_secretsmanager_secret_version.db_password.secret_id
```

I was running into a condition where, when this referenced the `aws_secretsmanager_secret`, the RDS instance would be one version behind on the secret version. I haven't been involved in the conversations around these improvements so feel free to push back on this suggestion if there's something I'm misunderstanding though!